app/assets/javascripts/app/controllers/recovery_page_controller.js

```diff
@@ -0,0 +1,93 @@
+import Modal from '../components/modal';
```

Comment:
Maybe, if people like the naming convention!

In this case, the JS is 100% tied to this specific view. Is it responsible for responding to events and potentially re-rendering DOM elements based on the outcome of those events. This is more or less analogous to a view controller.

These are frequently just called 'views' in JS apps, but that term obviously has another meaning in Rails apps